src/kvstore/Part.cpp

```diff
@@ -409,6 +410,19 @@ bool Part::preProcessLog(LogID logId, TermID termId, ClusterID clusterId, const
         if (ts > startTimeMs_) {
           LOG(INFO) << idStr_ << "preprocess add learner " << learner;
           addLearner(learner);
+
+          // persist the part info in case of storaged restarting
+          if (engine_ != nullptr) {
```

Comment:
`engine_` won't be empty here, it is ok to remove it

Reply:
OK